File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i = 0
        j = len(nums) - 1
        sum = None
        while sum!=target and i < len(nums) - 1:
            for k in range(0,j-i):
                sum = nums[i] + nums[j]
                if sum != target:
                    j-=1
                elif sum == target:
                    return[i, j]
            i+=1
            j = len(nums) - 1
        return []

File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Create a hash map to store the number and its corresponding index
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            
            // If the complement exists in the map, we found our pair!
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            
            // Otherwise, store the current number and its index in the map
            map.put(nums[i], i);
        }
        
        // In case no solution is found (though the problem guarantees one)
        throw new IllegalArgumentException("No two sum solution found");
    }
}